src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +696,99 @@ export class ResidualHeapSerializer {
     return t.arrayExpression(initProperties);
   }
 
+  _getPropertyValue(object: any, key: string, serialzeValue: boolean) {
+    if (object != null && object.has(key) === true) {
+      let val = object.get(key);
+
+      if (val !== undefined) {
+        let descriptor = val.descriptor;
+
+        if (descriptor !== undefined) {
+          let descriptorValue = descriptor.value;
+
+          if (descriptorValue !== undefined) {
+            return serialzeValue === true ? this.serializeValue(descriptorValue) : descriptorValue;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  _serializeValueReactElement(val: ObjectValue): BabelNodeExpression {
+    let objectProps = val.properties;
+
+    // we do this so they don't get emited
+    this.serializedValues.add(val.$Prototype);
+    let $$typeof = this._getPropertyValue(objectProps, "$$typeof", false);
+    if ($$typeof !== null) {
+      this.serializedValues.add(($$typeof: any));
+      if ($$typeof.$Description != null) {
+        this.serializedValues.add(($$typeof.$Description: any));
+      }
+    }
+    let _owner = this._getPropertyValue(objectProps, "_owner", false);
+    if (_owner !== null) {
+      this.serializedValues.add((_owner: any));
+    }
+    let typeValue = this._getPropertyValue(objectProps, "type", true);
+    let keyValue = this._getPropertyValue(objectProps, "key", true);
+    let refValue = this._getPropertyValue(objectProps, "ref", true);
+    let propsValue = this._getPropertyValue(objectProps, "props", false);
+
+    if (propsValue !== null) {
+      this.serializedValues.add((propsValue: any));
+    }
+    invariant(typeValue !== null, "JSXElement type of null");
+
+    let identifier = convertExpressionToJSXIdentifier(typeValue);
+    let attributes = [];
+    let children = [];
+
+    if (keyValue !== null && keyValue.type !== "NullLiteral") {
+      attributes.push((0, convertKeyValueToJSXAttribute)("key", keyValue));
+    }
+
+    if (refValue !== null && refValue.type !== "NullLiteral") {
+      attributes.push((0, convertKeyValueToJSXAttribute)("ref", refValue));
+    }
+
+    if (propsValue !== null && propsValue.properties !== undefined) {
+      for (let [key, propertyBinding] of (propsValue: any).properties) {
+        let desc = propertyBinding.descriptor;
+        if (desc === undefined) continue; // deleted
+
+        if (key === "key" || key === "ref") {
+          throw new Error(key + " is a reserved prop name");
```

Comment:
Don't throw an error, call `logError`, preferably in the visitor, but here would be okay too. Make the message a bit more descriptive (prop name for JSX object).